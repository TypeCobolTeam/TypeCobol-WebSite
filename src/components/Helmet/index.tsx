import * as React from "react"
import Helmet from "react-helmet"

const CHelmet: React.StatelessComponent = () => {
  return (
    <Helmet
      defaultTitle="TypeCobol"
      titleTemplate="TypeCobol - %s"
      meta={[
        {
          name: "description",
          content: "Official Website for TypeCobol",
        },
        {
          name: "keywords",
          content: "cobol, language, parser, ibm",
        },
        {
          name: "og:type",
          content: "website",
        },
      ]}
    />
  )
}

export default CHelmet
