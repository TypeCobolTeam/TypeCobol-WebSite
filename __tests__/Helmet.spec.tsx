import * as React from "react"

// @ts-ignore
import Helmet from "@components/Helmet"
import { mount } from "enzyme"
import { Helmet as Helm } from "react-helmet"

// eslint-disable-next-line
let wrapper
let helmet

beforeAll(() => {
  wrapper = mount(<Helmet title="TypeCobol" lang="fr" />)
  helmet = Helm.peek()
})

describe("Helmet", () => {
  it("sets typecobol as page title", () => {
    expect(helmet.title).toMatch(/typecobol/i)
  })
  it("has necessary keywords for SEO", () => {
    expect(helmet.metaTags).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          name: "description",
        }),
      ])
    )
    expect(helmet.metaTags).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          name: "keywords",
        }),
      ])
    )
    expect(helmet.metaTags).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          name: "og:type",
        }),
      ])
    )
  })
})
