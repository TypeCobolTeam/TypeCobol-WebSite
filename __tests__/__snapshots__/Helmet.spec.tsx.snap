// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helmet renders correctly 1`] = `ReactWrapper {}`;
